reproserver/proxy.py

```diff
@@ -62,6 +62,25 @@ def get(self):
         return self.finish('Ok')
 
 
+
+class SubdirRewriteMixin:
+    def set_status(self, code, reason=None):
+        if code >= 300 and code < 400:
+            logger.info("Rewrite status {0} {1} -> 200 OK, add as headers".format(code, reason))
+            super(SubdirRewriteMixin, self).set_status(200, "OK")
+            self.set_header("x-redirect-status", str(code))
+            self.set_header("x-redirect-statusText", reason)
+        else:
+            super(SubdirRewriteMixin, self).set_status(code, reason)
+
+    def set_header(self, name, value):
+        if name and name.lower() == "location":
+            name = "x-orig-location"
+            logger.info("Rewrite location -> x-orig-location")
```

Comment:
Are you sure this doesn't need a `return` or a `else`? Doesn't it still set `location` in addition to `x-orig-location`?